### PR TITLE
Add Facet mosaic

### PR DESCRIPTION
**Pull request by PaulZC**
This is huge... (Sorry!)

This PR adds support for the Facet mosaic (X5, L-Band).
It is working well on the original v1.0. It has not yet been tested on the new v1.1 PCB.

Enjoy!
Paul

### CODE REVIEW

File: Firmware/RTK_Everywhere/GNSS.ino [hunk 1]
```diff
@@ -107,12 +131,18 @@ bool gnssIsSurveyComplete()
...
-            int elapsedSeconds = (millis() - um980BaseStartTimer) / 1000;
+            int elapsedSeconds = (millis() - autoBaseStartTimer) / 1000;
 
             if (elapsedSeconds > settings.observationSeconds)
                 return (true);
             return (false);
         }
+        else if (present.gnss_mosaicX5)
+        {
+            // Bit 6: Set if the user has entered the command setPVTMode, Static, , auto
```
Comment (nseidle) [inline Firmware/RTK_Everywhere/GNSS.ino:142]: Dual comma typo

File: Firmware/RTK_Everywhere/GNSS.ino [hunk 2]
```diff
@@ -130,6 +160,11 @@ bool gnssSurveyInReset()
             // Put UM980 into rover mode to cancel base averaging mode
             return (um980SetModeRoverSurvey());
         }
+        else if (present.gnss_mosaicX5)
+        {
+            // Put mosaicX5 into rover mode to cancel base averaging mode
+            return (mosaicX5SetModeRoverSurvey());
```
Comment (nseidle) [inline Firmware/RTK_Everywhere/GNSS.ino:166]: The function to put the mosaic into Rover mode is ModeRoverSurvey()? Why the Survey. I'll keep reading...

Comment (nseidle) [reply Firmware/RTK_Everywhere/GNSS.ino:166]: Would ModeSurveyReset be more fitting?

File: Firmware/RTK_Everywhere/GNSS.ino [hunk 3]
```diff
@@ -1287,6 +1644,150 @@ bool gnssIsBlocking()
...
+    return false;
+}
+
+bool gnssStandby()
+{
+    if (online.gnss == true)
+    {
+        if (present.gnss_zedf9p)
+        {
+            return true; // TODO - this would be a perfect plave for Save-On-Shutdown
```
Comment (nseidle) [inline Firmware/RTK_Everywhere/GNSS.ino:1741]: Typo

**Comment by nseidle** [comment]
Looks good to me! Merge at your convenience.

**Review by nseidle — APPROVED** [review]

File: Firmware/RTK_Everywhere/GNSS.ino (hunk 1 above)
Comment (PaulZC) [reply Firmware/RTK_Everywhere/GNSS.ino:142]: The dual comma is correct...

File: Firmware/RTK_Everywhere/GNSS.ino (hunk 3 above)
Comment (PaulZC) [reply Firmware/RTK_Everywhere/GNSS.ino:1741]: Darn my dyslexic fingers... Fixing!

File: Firmware/RTK_Everywhere/GNSS.ino (hunk 2 above)
Comment (PaulZC) [reply Firmware/RTK_Everywhere/GNSS.ino:166]: ```mosaicX5SetModeRoverSurvey``` is the X5 version of ```gnssSurveyInReset```
```gnssSurveyInReset``` is only called during the state transition from ```STATE_BASE_TEMP_SURVEY_STARTED``` to ```STATE_ROVER_NOT_STARTED```. Comments say "Survey-In took more than %d minutes. Returning to rover mode"
Yeah. Agreed. The name needs to change. Let's go with ```mosaicX5SurveyReset```.

**Comment by PaulZC** [comment]
Excellent - thanks!

I'll test it on the new v1.1 hardware and merge later today.

**Comment by PaulZC** [comment]
Working well on Facet mosaic v11! Merging...

**Pull request merged**